src/main/java/Storage.java

```diff
@@ -0,0 +1,103 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class Storage {
+    private static String FILE_PATH = "data/duke.txt";
+
+    /**
+     * Adds all tasks found in storage to input tasks.
+     *
+     * @param tasks TaskList to add read tasks to
+     * @throws FileNotFoundException Thrown when file is not found
+     */
+    public static void readFile(TaskList tasks) throws IOException {
+        try {
+            File f = new File(FILE_PATH);
+            Scanner s = new Scanner(f);
+            while (s.hasNext()) {
+                String line = s.nextLine();
+                Parser parse = new Parser(line, true);
+                Task task = parse.createTask();
+                tasks.add(task);
+            }
+        } catch (FileNotFoundException e) {
+            new File("data").mkdir(); // Create directory
+            deleteContent(); // Create file
+            readFile(tasks);
+        }
+    }
+
+    /**
+     * Writes task to storage.
+     *
+     * @param task Task to store in storage
+     * @throws IOException Thrown when I/O error occurs
+     */
+    public static void addTask(String task) throws IOException {
+        FileWriter fw = new FileWriter(FILE_PATH, true);
+        fw.write("0" + " " + task + "\n");
+        fw.close();
+    }
+
+    /**
+     * Delete task from storage.
+     *
+     * @param index Index of task to remove from storage
+     * @throws IOException Thrown when I/O error occurs
+     */
+    public static void deleteTask(int index) throws IOException {
```

Comment:
Perhaps it could be better to have an in-memory Task List so that your deleteTask and doTask commands don't have to read the Tasks from disk, then modify accordingly, then save it back to disk. They can just modify the in memory Task List, then save it back to the disk directly. This should be more efficient.